src/time_steppers/backwardEuler.cpp

```diff
@@ -84,7 +85,7 @@ double backwardEuler::newtonMethod(double dt) {
 }
 
 
-void backwardEuler::lineSearch(double dt) {
+void backwardEuler::lineSearch_gs(double dt) {
```

Comment:
Instead of having multiple functions, I think a `lineSearch` class that is a member of `backwardEuler` would be a better idea. Can then use inheritance to easily expand to different line search methods later.